### PR TITLE
Release a new version with GitHub Actions

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

If releases are performed through an automated task, mistakes can be reduced and more validation can be added.

Modifications:

- Add `tag-new-release.yml` to tag a new version with GitHub Actions dispatch UI.

Result:

Automated release process

### CODE REVIEW

**Comment by ikhoon** [comment]
Give credit to @AmosDoan since the workflow was inspired by his work at LINE.

**Comment by ikhoon** [comment]
The job was tested at the forked repo. 
- https://github.com/ikhoon/armeria/actions/runs/6653695760

File: .github/workflows/tag-new-release.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            exit 1
+          fi
+          if [[ ! "${{ inputs.next_version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "Error: 'next_version' should be in SemVer format."
+            exit 1
+          fi
+
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
```
Comment (ikhoon) [inline .github/workflows/tag-new-release.yml:36]: PAT is required to trigger another workflow. https://stackoverflow.com/a/60418414/1736581

File: .github/workflows/tag-new-release.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+name: Release a new Armeria version
+on:
+  workflow_dispatch:
```
Comment (jrhee17) [inline .github/workflows/tag-new-release.yml:3]: I'm not really familiar with this. So is there a UI where we can supply/trigger these inputs?

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by jrhee17 — APPROVED** [review]
Thanks @ikhoon 🙇 👍 🙇

File: .github/workflows/tag-new-release.yml (hunk 2 above)
Comment (ikhoon) [reply .github/workflows/tag-new-release.yml:3]: Right. If you click the workflow in https://github.com/line/armeria/actions, we can see the input from the right upper corner.
image

**Review by trustin — APPROVED** [review]
:+1::+1::+1:

**Comment by ikhoon** [comment]
Thanks for the quick review. 🙇‍♂️🙇‍♂️

**Comment by ikhoon** [comment]
The job failed because `gpgconf` wasn't installed on self-hosted runners.
```
Unable to locate executable file: gpgconf. ...
```
`gnupg2-full` has been installed to install `gpgconf`.
```
$ sudo dnf install --allowerasing gnupg2-full
```

**Pull request merged**